src/libcollectdclient/network_parse.c

```diff
@@ -154,11 +154,11 @@ static int parse_string(void *payload, size_t payload_size, char *out,
                         size_t out_size) {
   char *in = payload;
 
-  if ((payload_size < 1) || (in[payload_size - 1] != 0) ||
+  if ((payload_size < 1) || (in[payload_size - 1] != '\0') ||
```

Comment:
Nit, at your option: Personally I prefer `0` over `'\0'`, but there is plenty of precedence for the latter.